libcxx/include/__ranges/chunk_by_view.h

```diff
@@ -205,17 +205,19 @@ namespace views {
 namespace __chunk_by {
 struct __fn {
   template <class _Range, class _Pred>
-  _LIBCPP_NODISCARD_EXT _LIBCPP_HIDE_FROM_ABI constexpr auto operator()(_Range&& __range, _Pred&& __pred) const
-      noexcept(noexcept(/**/ chunk_by_view(std::forward<_Range>(__range), std::forward<_Pred>(__pred))))
-          -> decltype(/*--*/ chunk_by_view(std::forward<_Range>(__range), std::forward<_Pred>(__pred))) {
-    return /*-------------*/ chunk_by_view(std::forward<_Range>(__range), std::forward<_Pred>(__pred));
+  _LIBCPP_NODISCARD_EXT _LIBCPP_HIDE_FROM_ABI static constexpr auto operator()(
+      _Range&& __range,
+      _Pred&& __pred) noexcept(noexcept(chunk_by_view(std::forward<_Range>(__range), std::forward<_Pred>(__pred))))
+      -> decltype(/*-----------------*/ chunk_by_view(std::forward<_Range>(__range), std::forward<_Pred>(__pred))) {
+    return /*------------------------*/ chunk_by_view(std::forward<_Range>(__range), std::forward<_Pred>(__pred));
   }
 
   template <class _Pred>
     requires constructible_from<decay_t<_Pred>, _Pred>
-  _LIBCPP_NODISCARD_EXT _LIBCPP_HIDE_FROM_ABI constexpr auto operator()(_Pred&& __pred) const
-      noexcept(is_nothrow_constructible_v<decay_t<_Pred>, _Pred>) {
-    return __range_adaptor_closure_t(std::__bind_back(*this, std::forward<_Pred>(__pred)));
+  _LIBCPP_NODISCARD_EXT _LIBCPP_HIDE_FROM_ABI static constexpr auto
+  operator()(_Pred&& __pred) noexcept(is_nothrow_constructible_v<decay_t<_Pred>, _Pred>) {
+    constexpr auto __self = __fn{};
+    return __range_adaptor_closure_t(std::__bind_back(__self, std::forward<_Pred>(__pred)));
```

Comment:
- `std::__bind_back(*this, std::forward<_Pred>(__pred))` won't compile because `this` is not available in a `static` function.
- `std::__bind_back(operator(), std::forward<_Pred>(__pred))` won't compile because the compiler can't determine which overload of `operator()` to use.

Please let me know if there's a better solution!

Reply:
> The solution here is to initialize the function object `__fn` as a `constexpr` variable, which will be evaluated at compile time, and pass it to `std::__bind_back`. Please let me know if there's a better solution!

This looks good to me. [P2714R1](https://wg21.link/p2714r1) (adopted for C++26) exactly adds such things.

Reply:
> The solution here is to initialize the function object `__fn` as a `constexpr` variable, which will be evaluated at compile time, and pass it to `std::__bind_back`. Please let me know if there's a better solution!

Perhaps we can "pass" the functor as a template argument in the same way as [P2714R1](https://wg21.link/p2714r1) (adopted for C++26).

Reply:
I wonder whether we should change this place. We need a `*this` pointer so what is the benefit of this change?

Reply:
> I wonder whether we should change this place. We need a `*this` pointer so what is the benefit of this change?

This change allows the first overload of `operator()` to be `static`. We just need to figure out how to pass `__fn` to `std::__bind_back` here.

Reply:
We can't this is a C++23 feature and P2714 is a C++26 feature.

Reply:
@philnik777 @frederick-vs-ja What do you think?

Reply:
> If P2714R1 is implemented

> We can't this is a C++23 feature and P2714 is a C++26 feature.

Oh, but IIUC we don't need to be waiting for P2714R1 being implemented. `__bind_back` is not a standard future and it seems OK to make it equivalent to C++26 `std::bind_back` in C++20 mode.

Reply:
The  `__bind_back` implementation detail is used C++20 so that "runs" into the same issue. I really want to avoid this change in this PR. We can make a separate PR where we add a `__bind_back` for C++23 which does that. However in that case I want to see a measurable performance improvement before adding that.

Reply:
> The `__bind_back` implementation detail is used C++20 so that "runs" into the same issue. I really want to avoid this change in this PR. We can make a separate PR where we add a `__bind_back` for C++23 which does that. However in that case I want to see a measurable performance improvement before adding that.

Sounds good to me. Thanks for reviewing this PR!

Reply:
The changes on this file have been reverted.